server/resources/TemplateResources.py

```diff
@@ -1,22 +1,100 @@
+                'message': f'Template {template.id} deleted'
+            }, 200
+        else:
+            return {
+                'message': 'Template not found'
+            }, 404
+
+    @jwt_required
+    def get(self, id):
+        current_user = UserModel.find_by_id(get_jwt_identity())
+        template = current_user.templates.filter_by(id=id).first()
+        return {
+            'template': serializableTemplate(template)
+        }, 200
+
+    @jwt_required
+    def put(self, id):
+        data = parser.parse_args()
+        current_user = UserModel.find_by_id(get_jwt_identity())
+        template = current_user.templates.filter_by(id=id).first()
+        if data['name'] is not None: template.name = data['name']
+        if data['type'] is not None: template.type = data['type']
+        if data['subject'] is not None: template.subject = data['subject']
+        if data['body'] is not None: template.body = data['body']
+        try:
+            template.update()
+            return {
+                'template': serializableTemplate(template)
+            }, 200
+        except ValueError as e:
+            return {
+              'message': str(e)
+            }, 400
+
+
+class Templates(Resource):
     @jwt_required
     def get(self):
-        email_templates = EmailTemplateModel.get_templates_by_owner_id(get_jwt_identity())
+        current_user = UserModel.find_by_id(get_jwt_identity())
+        templates = [ serializableTemplate(t) for t in current_user.templates.all() ]
+        return {
+            'templates': templates
+        }, 200
+
+    @jwt_required
+    def post(self):
+        data = parser.parse_args()
+        print(data)
+        current_user = UserModel.find_by_id(get_jwt_identity())
         try:
+            template = EmailTemplateModel(
+                name=data['name'],
+                type=data['type'],
+                subject=data['subject'],
+                body=data['body'],
+                owner_id=current_user.id
+            )
+            template.save_to_db()
             return {
-                'email_templates': [
-                    template.to_dict(rules=('-steps.email_template', '-steps.campaign', '-owner'))
-                    for template in email_templates
-                ]
+                'template': serializableTemplate(template)
             }, 200
-        except:
-            return {'message': 'Something went wrong'}, 500 
\ No newline at end of file
+        except ValueError as e:
+            return {
+              'message': str(e)
+            }, 400
+        except Exception as e:
+            return {
+                'message': str(e)
+            }, 500
+    
+
+def serializableTemplate(template):
+    ret = {
+        'id': template.id,
+        'name': template.name,
+        'type': template.type,
+        'subject': template.subject,
+        'body': template.body,
+        'dateCreated': str(template.date_created)
+    }
+    return ret
```

Comment:
I am using sqlalchemy-serializer library to serialize db Models

You can find the documentation here:
https://github.com/n0nSmoker/SQLAlchemy-serializer

It's convenient and easy to use, but make sure to read the rules, you can also check my examples